smac/facade/hydra_facade.py

```diff
@@ -33,56 +86,115 @@ class Hydra(object):
         loggs information about used resources
     solver : SMBO
         handles the actual algorithm calls
-    runhistory : RunHistory
+    rh : RunHistory
         List with information about previous runs
-    trajectory : list
+    portfolio : list
         List of all incumbents
+
     """
 
     def __init__(self,
-                 scenario: Scenario,
+                 scenario: typing.Type[Scenario],
                  n_iterations: int,
+                 val_set: str='train',
+                 incs_per_round: int=1,
+                 n_optimizers: int=1,
+                 rng: typing.Optional[typing.Union[np.random.RandomState, int]]=None,
                  run_id: int=1,
                  tae: typing.Type[ExecuteTARun]=ExecuteTARunOld,
                  **kwargs):
-        """Constructor
+        """
+        Constructor
 
         Parameters
         ----------
         scenario : ~smac.scenario.scenario.Scenario
             Scenario object
         n_iterations: int,
             number of Hydra iterations
-        """
+        val_set: str
+            Set to validate incumbent(s) on. [train, valX].
+            train => whole training set,
+            valX => train_set * 100/X
+        incs_per_round: int
+            Number of incumbents to keep per round
+        n_optimizers: int
+            Number of optimizers to run in parallel per round
+        rng: int/np.random.RandomState
+            The randomState/seed to pass to each smac run
+        run_id: int
+            run_id for this hydra run
+        tae: ExecuteTARun
```

Comment:
can we also provide a function handle here? similar as in the SMAC facade?

Reply:
Will do.

Reply:
Now the ExecuteTARunHydra can also handle function handles (which also inherit from ExecuteTARun)